### PR TITLE
Replace getExperiment* and testExperiment* functions with getCrossAssociation

### PR DESCRIPTION
**Pull request by theotimeee**
Create new function `getCrossAssociation` to replace the following :

- getExperimentCrossAssociation
- getExperimentCrossCorrelation
- testExperimentCrossAssociation
- testExperimentCrossCorrelation

New function `getCrossAssociation` makes the significance test as default `(test_significance` default value is `TRUE`)

### CODE REVIEW

File: R/deprecate.R [hunk 1]
```diff
@@ -56,3 +56,83 @@ setMethod("taxonomyTree", signature = c(x = "SummarizedExperiment"),
...
+
+#' @rdname deprecate
+#' @export
+setMethod("getExperimentCrossAssociation", 
+            signature = c(x = "MultiAssayExperiment"),
+            function(x, ...){
+                .Deprecated(msg = paste0("'getExperimentCrossAssociation' is ",
+                                        "deprecated. Use ", 
+                                        "'getCrossAssociation' instead."))
+                getExperimentCrossAssociation(x, ...)
```
Comment (TuomasBorman) [inline R/deprecate.R]: These functions are calling themselves

Comment (TuomasBorman) [reply R/deprecate.R]: Not the one that is new (getCrossAssociation)

**Comment by TuomasBorman** [comment]
Looks good, the idea of this PR was to remove additional functions that we have only one function for cross-correlation. Moreover, the name of function was shortened

**Comment by TuomasBorman** [comment]
test_significance could still be FALSE by default. The function uses base (or stats) R by default for calculating cross-correlation. The function that calculates correlation without significance testing is much faster

File: R/getCrossAssociation.R [hunk 2]
```diff
@@ -277,12 +274,12 @@ setMethod("getExperimentCrossAssociation", signature = c(x = "MultiAssayExperime
     }
 )
 
-#' @rdname getExperimentCrossAssociation
-#' @aliases getExperimentCrossCorrelation
+#' @rdname getCrossAssociation
+#' @aliases getCrossCorrelation
```
Comment (TuomasBorman) [inline R/getCrossAssociation.R]: This line should be removed. There is no function called getCrossCorrelation

File: pkgdown/_pkgdown.yml [hunk 3]
```diff
@@ -46,7 +46,7 @@ reference:
   - getPrevalence
   - getPrevalentAbundance
   - getTopTaxa
-  - getExperimentCrossCorrelation
```
Comment (TuomasBorman) [inline pkgdown/_pkgdown.yml:50]: getCrossAssociation

**Review by TuomasBorman — APPROVED** [review]

**Comment by antagomir** [comment]
Some conflicts..?

**Comment by antagomir** [comment]
Otherwise seems fine

**Pull request merged**